server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1032,7 +1032,7 @@ private IndexingStrategy planIndexingAsPrimary(Index index) throws IOException {
                 currentVersion = versionValue.version;
                 currentNotFoundOrDeleted = versionValue.isDelete();
             }
-            if (index.getIfSeqNo() != SequenceNumbers.UNASSIGNED_SEQ_NO && versionValue == null) {
+            if (index.getIfSeqNo() != SequenceNumbers.UNASSIGNED_SEQ_NO && currentNotFoundOrDeleted) {
```

Comment:
Should we use `Versions.NOT_FOUND` instead of `currentVersion` (line 1039) to avoid leaking the version of the deleted document?

Reply:
I see your point, but I am not sure about this. If anyone uses both external versioning and seqno-based versioning (as in mixing them or being in a migration process between them), returning the currentVersion here makes sense. I guess the case is quite contrived but wanted to hear your opinion on this too.

Reply:
ok, fair point :).